sandbox/apparmor/notify/listener/listener_test.go

```diff
@@ -767,7 +767,6 @@ func (*listenerSuite) TestRunConcurrency(c *C) {
 	defer restoreOpen()
 
 	recvChan, sendChan, restoreEpollIoctl := listener.MockEpollWaitNotifyIoctl()
-	defer restoreEpollIoctl()
```

Comment:
why removing this?

Reply:
The restore function closes `sendChan` (since `sendChan` is passed to the test as read-only, we can't close it directly here), and the test checks a value which can only be checked after `sendChan` closes, so we need to manually call `restoreEpollIoctl()` before doing those final checks.

Reply:
but if the test panics, you'll miss the restore, you can reduce the situation mixing  the two approaches with a flag:

```
epollIoctlRestored := false
defer func() {
   if !epollIoctlRestored {
       restoreEpollIoctl()
   }
}()
...

epollIoctlRestored := true
restoreEpollIoctl()
```

Reply:
Great catch and suggestion, thanks! I'll add your fix.